apps/admin/src/pages/hackathons/hackathons.tsx

```diff
@@ -0,0 +1,100 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+
+import { insertHackathonSchema } from "@knighthacks/db";
+
+import { Button } from "~/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "~/components/ui/form";
+import { Input } from "~/components/ui/input";
+import { trpc } from "~/trpc";
+
+const insertHackathonFormSchema = insertHackathonSchema.omit({ id: true });
+
+type InsertHackathonFormValues = z.infer<typeof insertHackathonFormSchema>;
+
+function createHackathon() {
+  // Alternative way to mutate (for learning purposes)
+  // const hackathonRes1 = trpc.hackathons.createHackathon.useMutation().mutate({name: "bro", startDate: "bro", endDate: "bro"})
+}
+
+export function AddHackathonForm() {
+  const utils = trpc.useUtils();
+  const { mutate, isLoading } = trpc.hackathons.createHackathon.useMutation({
+    onSuccess: async () => {
+      await utils.hackathons.getCurrentHackathon.invalidate();
+      toast("Success!", { description: "Hackathon added" });
+    },
+  });
+  const form = useForm<InsertHackathonFormValues>({
+    resolver: zodResolver(insertHackathonFormSchema),
+    defaultValues: {
+      name: "",
+      startDate: "",
+      endDate: "",
+    },
+  });
+
+  const onSubmit: SubmitHandler<InsertHackathonFormValues> = (values) => {
+    mutate(values);
+  };
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="flex flex-col justify-center space-y-6"
+      >
+        <FormField
+          control={form.control}
+          name="name"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Hackathon Name</FormLabel>
+              <FormControl>
```

Comment:
This input should probably of `type="date"` whose value should be formatted as a string like this YYYY-MM-DD. But if you can't fix this atm, feel free to just open another pr to do so!